### ISSUE TITLE
Transition To Workbench in May

### ISSUE THREAD

**Issue by zkamvar** [datacarpentry/organization-genomics#152]
@datacarpentry/organization-genomics-maintainers

As I hope you are already aware, we are rolling out the new lesson infrastructure, [The Carpentries Workbench](https://carpentries.github.io/workbench/), across all of The Carpentries official lessons in early May 2023. This means that all Data Carpentry, Library Carpentry, and Software Carpentry lesson repositories will be modified to adopt the new infrastructure in the coming days.

You can follow the transition of this lesson repository at https://github.com/carpentries/lesson-transition/issues/54.

The transition has already taken place for several lessons, and so far the process has been running quite smoothly. You should see the transition take place with minimal disruption, but there are a few things that it is important for Maintainers to be aware of. 

Here is what you can expect to happen next:

1. Any open pull requests on the repository will be closed with an automated message.
2. The repository will be set to read-only mode for a brief period while the transition occurs.
3. The new repository structure and lesson site layout will then be applied.
4. To avoid anyone accidentally pushing the old commit history back to the repository, after the transition Maintainers will need to delete and replace any existing forks and local clones they have of the lesson repository, and confirm that they have done so by replying to this issue.

I will reply here before and after the transition has taken place. If you have any questions in the meantime, please reach out to the Curriculum Team by tagging us here, e.g. @datacarpentry/core-team-curriculum.

If you would like to read more about the new lesson infrastructure and the modified repository structure you can expect post-transition, I recommend [the Infrastructure episode of the Maintainer Onboarding curriculum](https://carpentries.github.io/maintainer-onboarding/05-infrastructure.html) and [the Workbench Transition Guide](https://carpentries.github.io/workbench/transition-guide.html), which includes a side-by-side comparison of various elements of the old and new infrastructures.

**Comment by zkamvar** [comment]
We are all set to transition this lesson tomorrow. I have fixed all the issues that were identified and you can see the preview here: https://fishtree-attempt.github.io/organization-genomics/

**Comment by zkamvar** [comment]
This lesson will be converted to use [The Carpentries Workbench][workbench]
To prevent accidental reversion of the changes, we are temporarily revoking
write access for all collaborators on this lesson:

 - [ ] @naupaka (push)
 - [ ] @crazyhottommy (push)
 - [x] @JCSzamosi (push)
 - [ ] @hoytpr (push)
 - [ ] @ameynert (push)
 - [ ] @JasonJWilliamsNY (push)
 - [ ] @aschuerch (push)
 - [ ] @elizabethmcd (push)
 - [x] @hidyverse (push)

If you no longer wish to have write access to this repository, you do not
need to do anything further.

1. What you can expect from the transition 📹: https://carpentries.github.io/workbench/beta-phase.html#beta
2. How to update your local clone 💻: https://carpentries.github.io/workbench/beta-phase.html#updating-clone
3. How to update (delete) your fork (if you have one) 📹: https://carpentries.github.io/workbench/faq.html#update-fork-from-styles

If you wish to regain write access, please re-clone the repository on your machine and
then comment here with `I am ready for write access :rocket:` and the
admin maintainer of this repository will restore your permissions.

If you have any questions, please reply here and tag @zkamvar

[workbench]: https://carpentries.github.io/workbench

**Comment by zkamvar** [comment]
The deed is done. The infrastructure takes a few minutes to bootstrap and cache the packages for the lesson build. Once the build is done, I will switch github pages to deploy from the `gh-pages` branch and you will have your workbench lesson. 

Thank you all for your enthusiasm and your patience!

**Comment by zkamvar** [comment]
The Workbench version is now live: https://datacarpentry.github.io/organization-genomics/

In addition, here is [map of commits that were changed during the transition](https://github.com/carpentries/lesson-transition/blob/release_dc/organization-genomics/release/datacarpentry/organization-genomics-commit-map.hash)

**Comment by JCSzamosi** [comment]
I am ready for write access :rocket:

**Comment by zkamvar** [comment]
@JCSzamosi, access granted 🎉

**Comment by JCSzamosi** [comment]
Thanks for all your work and exceptionally clear communication throughout this process @zkamvar !

**Comment by zkamvar** [comment]
It's my pleasure!

**Comment by hoytpr** [comment]
It looks great. The primary actionable item might be that at the end of the lesson, the "next" arrow cycles back to what looks like the beginning of the workshop. The "Summary" starts the workshop again.

**Comment by hidyverse** [comment]
I am ready for write access :rocket:

**Comment by zkamvar** [comment]
Access granted @hidyverse (and hella props for the awesome handle)!

**Comment by zkamvar** [comment]
> It looks great. The primary actionable item might be that at the end of the lesson, the "next" arrow cycles back to what looks like the beginning of the workshop. The "Summary" starts the workshop again.

@hoytpr, This is not something that can be fixed in individual lessons, it is an infrastructure issue. Could you open an issue in https://github.com/carpentries/workbench/issues ? 

Additionally, would you say `I am ready for write access :rocket:` or do you still need to update your local clone?

**Comment by hoytpr** [comment]
Hi @zkamvar,

I'm not ready for write access just yet. I have retired and the computer situation is in flux. I will see if my home setup still works ASAP.

Peter R. Hoyt, PhD
Emeritus Associate Professor
Oklahoma State University
Biochemistry and Molecular Biology
110FC HBRC
405-774-6206
________________________________

**Comment by zkamvar** [comment]
> I'm not ready for write access just yet. I have retired and the computer situation is in flux. I will see if my home setup still works ASAP.

No worries and no rush! The main concern is to prevent anyone who has an existing clone from accidentally force-pushing changes. If you are going to work from a fresh clone, you should be fine.

**Comment by naupaka** [comment]
I am ready for write access :rocket:

**Comment by tobyhodges** [comment]
Hey @naupaka thanks for following up. To shore up the security of the lesson repositories, the Curriculum Team would like to use this opportunity to tidy up the access levels we are providing to community members across the various repos. According to our records, you're not a Maintainer on this lesson, but I think you had previously been granted write access to the repository in your capacity as a DC Genomics Curriculum Advisor. 

For now, we are going to provide write access only to Maintainers - though that decision will be open to review, so please let us know (@ me or datacarpentry/core-team-curriculum) if this interferes with the CAC's ability to perform their duties.

**Comment by naupaka** [comment]
@tobyhodges no worries! happy to have less to worry about 😅

**Comment by JCSzamosi** [comment]
Does anyone have any objection to closing this issue? It seems like @hoytpr has been granted write access and I don't see any obvious outstanding issues. @zkamvar @tobyhodges

**Comment by zkamvar** [comment]
Thank you for asking this, @JCSzamosi

@hoytpr has Triage access, but he does not yet have Write access. He is able to merge pull requests, but he needs to use his own fork if he wants to make changes at the moment. 

Once he says `I am ready for write access :rocket:`,then I will elevate his access.

**Comment by ErinBecker** [comment]
Hi @hoytpr - You have not yet responded here to report that you completed the steps listed by @zkamvar above to regain your Maintainer access to this repository. The lesson infrastructure transition required a second Maintainer team to be temporarily created for all lessons, which can only be removed when all Maintainers have confirmed that they are ready to have their access restored. Maintenance of both teams is not sustainable for The Carpentries in the long run. At the beginning of **November 2023** (note the deadline has been extended) we will remove access for Maintainers who have not taken the necessary action to continue working on the repository.

The Curriculum Team is truly grateful for everything you have already done as a Maintainer and we do not want to lose you! But we also understand that circumstances change and you may no longer have capacity to continue in your role. If you wish to retain your access and keep supporting the community, please follow the steps above then confirm that you are ready by commenting I am ready for write access 🚀 on this thread before the end of your day on **Wednesday 1st November.**

**Comment by ErinBecker** [comment]
Hi @hoytpr  - You no longer have Maintainer access to this repository. Please see the comment above for details. If you would like to restore your access, please re-clone the repository on your machine and then comment here with I am ready for write access 🚀 and the admin maintainer of this repository will restore your permissions. If you have any questions, please don’t hesitate to get in touch [curriculum@carpentries.org](mailto:curriculum@carpentries.org)

To the remaining Maintainers for this lesson - the Curriculum Team will be prioritizing onboarding of new Maintainers for lessons that have lost Maintainers in our next round of onboarding. Please don’t hesitate to reach out if your lesson needs additional support in the meantime - [curriculum@carpentries.org](mailto:curriculum@carpentries.org).

I will close this issue now but will still be notified of any replies on the thread in future.

**Comment by hoytpr** [comment]
@ErinBecker and @zkamvar, 
My sincere apologies. Life happens and I've not regained my full abilities since I was ill 18 months ago. I want to continue with Carpentries, but perhaps it's better to put me on "temporary leave" while I continue to heal (which is going, but going slow). 

The transition to workbench simply came at a bad time when taking on new challenges is still difficult. I'm still myself (take that as good or bad), but getting caught up with home repairs and even trivial stuff is slow. Happy to speak more about it here, or by email, or phone. But I have to keep focus on my life. 

I didn't make this decision until just today when I realized it was better for everyone for me to stay in touch, but let new, probably better people step up. 

Please don't remove me from too many email lists (or slack channels, or whatever). I want to return as soon as possible. And I don't want to lose all the great friends and family in the Carpentries (I'd miss you all too much). I'll have to be the old uncle who "goes away" for a while. 
Peter (@hoytpr)

**Comment by JCSzamosi** [comment]
Please don't worry too much @hoytpr ! I think @hidyverse and I can hold down the fort here while you take care of yourself. I hope to see you around elsewhere!

**Comment by hoytpr** [comment]
Thanks @JCSzamosi, with a little luck, I'll be back here or somewhere around doing some good for data science education.

**Comment by tobyhodges** [comment]
You've nothing to apologise for, @hoytpr. I'm so sorry to hear you've been unwell and that the recovery is slow. Thank you for thinking it through, making a conscious decision to step back, and taking the time to update us here. We'll be delighted to welcome you back when you are ready, but until then look after yourself.
